packages/next/server.js

```diff
@@ -1,17 +1,23 @@
-const serverExports = {
-  NextRequest: require('next/dist/server/web/spec-extension/request')
-    .NextRequest,
-  NextResponse: require('next/dist/server/web/spec-extension/response')
-    .NextResponse,
-  userAgentFromString: require('next/dist/server/web/spec-extension/user-agent')
-    .userAgentFromString,
-  userAgent: require('next/dist/server/web/spec-extension/user-agent')
-    .userAgent,
-}
+/* global URLPattern */
 
-if (typeof URLPattern !== 'undefined') {
-  // eslint-disable-next-line no-undef
-  serverExports.URLPattern = URLPattern
+const serverExports = {
+  NextRequest:
+    /*#__PURE__*/ require('next/dist/server/web/spec-extension/request')
+      .NextRequest,
+  NextResponse:
+    /*#__PURE__*/ require('next/dist/server/web/spec-extension/response')
+      .NextResponse,
+  userAgentFromString:
+    /*#__PURE__*/ require('next/dist/server/web/spec-extension/user-agent')
+      .userAgentFromString,
+  userAgent:
+    /*#__PURE__*/ require('next/dist/server/web/spec-extension/user-agent')
+      .userAgent,
+  URLPattern:
+    typeof EdgeRuntime === 'string'
```

Comment:
```suggestion
    typeof URLPattern !== 'undefined'
```
Should we check for URLPattern being present instead of the `EdgeRuntime` global?

Reply:
yeah we can do that. I wonder what more APIs we want to expose here. we can have streams here too, I guess

Reply:
ah I remember why I did it that way. `typeof URLPattern` will not allow dead code elimination!

Reply:
therefore, using `typeof EdgeRuntime === 'string'` will make the Edge compiler minify it to `URLPattern: globalThis.URLPattern` without importing the polyfills at all (which the edge runtime does not need)

Reply:
let's add it as a comment since it's difficult to notice